README.md

```diff
@@ -126,6 +127,18 @@ for read in R1 R2 I1 I2; do cat L*/${read}.fastq.gz > ./${read}.fastq.gz; done
 
 FASTQ files _must_ be BGZF compressed.
 
+All reads containing sample barcodes/indexes _must_ be of the same length.
```

Comment:
I think this actually confuses things.  I know what you're trying to say here because of the context of this PR, but are you really saying that if I have a fastq with read structure `8B+T` that it's now required that all reads from the matching fastq must be a fixed length?

I think what you really mean is that sample barcode segments in read structures must be of fixed lengths.

Reply:
No, I really do mean this, I think.  To infer from the input FASTQ the read length so I can make variable length sample barcode segments of fixed lengths, I am going to look at the first read in the FASTQ (not all), so. want the read length in the FASTQ to be same length.

Reply:
I am going to enforce that sample barcode segments must be fixed lengths